src/main/util.ts

```diff
@@ -45,3 +48,25 @@ export interface EntityList<T> {
     entities: T[];
     totalCount: number;
 }
+
+export function addClassMetadata<T>(key: Symbol, target: any, datum: T) {
+    const metadata = Reflect.getOwnMetadata(key, target) || [];
+    metadata.push(datum);
+    Reflect.defineMetadata(key, metadata, target);
+}
+
+export function getClassMetadata<T>(key: Symbol, target: any): T[] {
+    let result: T[] = [];
+    let proto = target;
+    while (proto !== Object.prototype) {
+        const ownMetadata: T[] = Reflect.getOwnMetadata(key, target) || [];
+        result = ownMetadata.concat(result);
+        proto = Object.getPrototypeOf(proto);
+    }
+    return result;
+}
+
+export function getBindingsMap(container: Container):
+    Map<interfaces.ServiceIdentifier<any>, interfaces.Binding<any>[]> {
+    return (container as any)._bindingDictionary._map;
```

Comment:
interesting to see to type inferring to `any`, but I assume type definition doesn't include `_bindingDictionary`

Reply:
Yep, unfortunately they don't expose internals, so this is my way of accessing their `private` stuff